ToukMultiplex/src/main/java/com/example/toukmultiplex/Reservation/ComputeReservationTimeArrival.java

```diff
@@ -0,0 +1,31 @@
+package com.example.toukmultiplex.Reservation;
+
+import com.example.toukmultiplex.Config.ReservationProperties;
+import org.springframework.stereotype.Service;
+
+import java.sql.Timestamp;
+import java.time.Instant;
+import java.util.Optional;
+
+@Service
+public class ComputeReservationTimeArrival {
+
+    private final ScreeningRepository screeningRepository;
+    private final ReservationProperties reservationProperties;
+
+    public ComputeReservationTimeArrival(ScreeningRepository screeningRepository, ReservationProperties reservationProperties) {
+        this.screeningRepository = screeningRepository;
+        this.reservationProperties = reservationProperties;
+    }
+
+    public Timestamp computeReservationTimeArrival(Long id){
+
+       Timestamp screeningTime;
+       Timestamp reservationTimeArrival = Timestamp.from(Instant.now());
+
+       Optional<Screening> screening = screeningRepository.findById(id);
+       screeningTime= screening.get().getScreeningDate();
+        reservationTimeArrival = Timestamp.from(screeningTime.toInstant().minusSeconds(reservationProperties.getExpireLimitOfAppearingInPersonMinutes()*60L));
```

Comment:
1. Dlaczego korzystasz z sql.Timestamp do przechowywania czasu?
2. Czy w serwisie można użyć innego api do przechowywania czasu?

Reply:
To chyba zła decyzja była której w trakcie pisania programu trochę załowałem. Spojrzałem że w Postresie jest
Timestamp, w javie Timestamp... ale teraz korzystał bym raczej z Instant